packages/generators/src/addon-generator.ts

```diff
@@ -4,6 +4,16 @@ import Generator from 'yeoman-generator';
 
 import { List } from './utils/scaffold-utils';
 
+// Helper to get the template-directory content
+
+const getFiles = (dir) => {
+    return fs.readdirSync(dir).reduce((list, file) => {
+        const filePath = path.join(dir, file);
+        const isDir = fs.statSync(filePath).isDirectory();
```

Comment:
Maybe we need handle this case better, in some edge cases here can be exception, for example permissions

Reply:
How do you see it? In fact, this is used by projects like [vite](https://github.com/vitejs/vite)
https://github.com/vitejs/vite/blob/978d991293f5b8e1a4197ac4e3aee4fa2e838a88/packages/create-app/index.js#L318

Reply:
Yep, used, but not handle exceptions, just add `try/catch` around `getFiles`, i.e. `try { getFiles() } catch (error) { logger.error(error); process.exit(2) }`